hbase-client/src/main/java/org/apache/hadoop/hbase/client/MutableRegionInfo.java

```diff
@@ -126,7 +127,9 @@ private static int checkReplicaId(int regionId) {
     this.replicaId = checkReplicaId(replicaId);
     this.offLine = offLine;
     this.regionName = RegionInfo.createRegionName(this.tableName, this.startKey, this.regionId,
-      this.replicaId, !this.tableName.equals(TableName.META_TABLE_NAME));
+      this.replicaId,
+      //1 is region id of FIRST_META_REGION_INFO
+      !(this.tableName.equals(TableName.META_TABLE_NAME) && regionId == 1));
```

Comment:
What is happening here? '1' is the id of the hardcoded single region hbase:meta table but is it going to be the region id when we split meta?

Reply:
Post split meta the region ids will be in the non-legacy format. This special case here is to handle the primordial meta region which is in legacy format. So we need to catch if the primordial meta region is being instantiated here and make sure it is created in the legacy format. 

Note I think I need to handle primordial meta region replicas here as well. (Do I need to support that?)